### PR TITLE
Seeding Cassandra and DMBS Benchmarking Tests

### PR DESCRIPTION
**Pull request by ahchung23**
### Changes to repo structure
**Seeding Updates**
- contains dataSampleGenerator folder housing files for creating a sample data .csv with 1M data points
  - npm script _"write:sampleData"_ generates sample .csv file for postgres/cassandra
- database folder renamed to seed_database
- now includes cassandra.cql and cassandra-sample.cql files for seeding cassandra database
  - npm script _"seed:cassandra"_ & _"seed:cassandraSample"_ can be used to seed cassandra database with real or sample data respectively
  - seeding is done through a series of cqlsh commands within .cql files (COPY method used to import data from .csv files)

**DMBS Benchmarking Tests**
- uses Jest test suite to benchmark and test Db core read queries
- contains test_db folder housing _cassandra.test.js_ & _postgres.test.js_
- each test file tests core read queries in cassandra and postgres for two main results:
  1. Confirms that the result object from query has correct properties and structure
  1. Confirms query time is on average under 50ms (for 1000 queries made within the last 10% of 10M dataset) 

### Questions
1. Do you have any suggestions on how I may improve my table structure (especially primary key design) within Cassandra to achieve faster seeding or faster queries? 
1. Do you see any problems with my DBMS Benchmarking tests?
1. Do you have any suggestions on other tests I should perform to achieve a more complete benchmarking?

Any comments and suggestions are greatly appreciated! :smiley_cat:

### CODE REVIEW

**Comment by asuluntas** [comment]
Very neat code for testing and in general. I learned a lot form your tests! 
Looks good to me.

**Comment by uabrami** [comment]
@ahchung23 Very clean code Andrew, I appreciate all the tests you wrote and your documentation. Looks good to merge.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+node_modules/
+s3Config.json
+sdcData.csv
+sdc-postgresql-data.csv
+sdc-cassandra-data.csv
+sdc-sample-postgresql-data.csv
+sdc-sample-cassandra-data.csv
```
Comment (therobinkim) [inline .gitignore]: not sure if it applies, but you could consider ignoring all csv files
`*.csv`

File: dataGenerator/s3Images.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const AWS = require('aws-sdk');
+AWS.config.loadFromPath('s3Config.json');
+const s3 = new AWS.S3();
+
+const params = {
+  Bucket: 'sdc-andrew-movie-posters',
```
Comment (therobinkim) [inline dataGenerator/s3Images.js]: i would eventually extract this and ln 21 out to a config file (or `process.env` variable)

File: dataGenerator/s3Images.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  MaxKeys: 800,
+};
+
+const listAllObjects = async () => {
+  const imgUrlArray = [];
+  try {
+    const data = await s3.listObjectsV2(params).promise();
+    data.Contents.forEach((elem) => {
```
Comment (therobinkim) [inline dataGenerator/s3Images.js]: i think this is better as `imgUrlArray = data.Contents.map(...)`

(bc `forEach` + `push` === `map`)

File: seed_database/seedPostgreSQL.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+const { Pool } = require('pg');
+const path = require('path');
+const pool = new Pool({
+  user: 'postgres',
+  host: '127.0.0.1',
+  database: 'sdcandrew',
+  password: '',
```
Comment (therobinkim) [inline seed_database/seedPostgreSQL.js]: would be nice to extract these config variables out

File: seed_database/seedPostgreSQL.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  port: 5432,
+});
+
+const seedPostgres = async () => {
+  const client = await pool.connect();
```
Comment (therobinkim) [inline seed_database/seedPostgreSQL.js:7]: if `pool.connect()` can be `await`ed, maybe put this in the `try`-`catch` block as well?

**Comment by therobinkim** [comment]
Great work!
For the future, would recommend smaller, more frequent PRs whenever possible

File: .gitignore (hunk 1 above)
Comment (ahchung23) [reply .gitignore]: Thanks, Robin! It does indeed apply and I'll definitely do that.

File: dataGenerator/s3Images.js (hunk 3 above)
Comment (ahchung23) [reply dataGenerator/s3Images.js]: Will do.

File: seed_database/seedPostgreSQL.js (hunk 5 above)
Comment (ahchung23) [reply seed_database/seedPostgreSQL.js:7]: I tried this method previously but had problems using the client to perform release or rollback queries within the catch block and finally block!

Most examples within the documentation also created the pool connection outside the try blocks but ended the pool connection within the blocks.
https://node-postgres.com/features/pooling

File: dataGenerator/s3Images.js (hunk 2 above)
Comment (ahchung23) [reply dataGenerator/s3Images.js]: I will look into using the dotenv package and a .env file to factor out these configuration variables!

File: seed_database/seedPostgreSQL.js (hunk 4 above)
Comment (ahchung23) [reply seed_database/seedPostgreSQL.js]: I will look into using the dotenv package and a .env file to factor out these configuration variables!

**Comment by ahchung23** [comment]
Thanks again for your input Robin!

**Pull request merged**